File: Lecture1/exercise/GreatCircle.java
public class GreatCircle{
    public static void main(String[] args){
        // Read latitude and longtitude (in degrees) from command line. Convert them to radians.
        double x1 = Math.toRadians( Double.parseDouble(args[0]) );
        double y1 = Math.toRadians( Double.parseDouble(args[1]) );
        double x2 = Math.toRadians( Double.parseDouble(args[2]) );
        double y2 = Math.toRadians( Double.parseDouble(args[3]) );
        
        // Radius of the earth
        double r = 6371.0;

        double sine = Math.sqrt( Math.pow(Math.sin( (x2-x1)/2 ), 2) + Math.cos(x1) * Math.cos(x2) * Math.pow(Math.sin( (y2-y1)/2 ), 2));
        double distance = 2 * r * Math.asin( sine );

        System.out.println(distance + " kilometers");
    }
}

File: Lecture1/c_repro/GreatCircle.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

void main(int argc, char* argv[]){
    float x1 = atof(argv[1]) * M_PI / 180.0;
    float y1 = atof(argv[2]) * M_PI / 180.0;
    float x2 = atof(argv[3]) * M_PI / 180.0;
    float y2 = atof(argv[4]) * M_PI / 180.0;

    printf("%f %f %f %f \n", x1, y1, x2, y2);

    float r = 6371.0;

    float sine = sqrt( pow(sin((x2-x1)/2),2) + cos(x1) * cos(x2) * pow(sin((y2-y1)/2),2));
    float distance = 2 * r * asin( sine );

    printf("%5.3f kilometers.\n",distance);

}